### PR TITLE
Feature: makes bulk actions from donation list table extensible

### PR DESCRIPTION
**Pull request by glaubersilva**
## Description

This PR adds a way to extend the default core bulk actions. This is useful for addons that need to implement custom bulk actions on the donation list table like in this case: https://github.com/impress-org/give-pdf-receipts/pull/247 

## Affects

Bulk actions from the donations list table.

## Testing Instructions

After pulling [this PR](https://github.com/impress-org/give-pdf-receipts/pull/247) and active the PDF Receipts add-on, follow these steps:

1. Go to Give > Donations and select a few donations by clicking on the checkbox aside from the ID column;
2. Select the "Download Receipts" on the bulk actions selector and click on the apply button;
3. You should be redirected to Give > Tools > PDF Export where the download will be handled.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204470020369035

### CODE REVIEW

File: src/Donations/resources/components/DonationsListTable.tsx [hunk 1]
```diff
@@ -132,6 +134,43 @@ const bulkActions: Array<BulkActionsConfig> = [
     },
 ];
 
+if (window.GiveDonations.pdfExportToolApiRoot) {
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/Donations/resources/components/DonationsListTable.tsx:137]: @kjohnson  I like this idea, however when the Core reads the window.giveDonationsTableBulkActions value, this is still empty and when it is updated on the addon, it does not reflect in the core... I tried to use use the userEffect hook but seems it's not possible to use it with global objects as a dependency.

Any ideas?

image

[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/Donations/resources/components/DonationsListTable.tsx:137]: @kjohnson I think I found a simple solution that consists of retrieving the addons bulk actions inside the `BulkActionSelect` component, so ignore my last message. =)

**Comment by glaubersilva** [comment]
@kjohnson Thanks for the insights! I implement your suggestion, this way was necessary to create [this new PR](https://github.com/impress-org/give-pdf-receipts/pull/247) and I also added you as the reviewer.

**Review by kjohnson — APPROVED** [review]

**Review by Benunc — APPROVED** [review]
This works in manual testing.

**Pull request merged**